crates/router/src/utils/currency.rs

```
@@ -224,33 +199,60 @@ async fn successive_fetch_and_save_forex(
     state: &SessionState,
     stale_redis_data: Option<FxExchangeRatesCacheEntry>,
 ) -> CustomResult<FxExchangeRatesCacheEntry, ForexCacheError> {
+    // spawn a new thread and do the api fetch and write operations on redis.
+    let forex_api_key = state.conf.forex_api.get_inner().api_key.peek();
+    if forex_api_key.is_empty() {
+        Err(ForexCacheError::ConfigurationError("api_keys not provided".into()).into())
+    } else {
+        let state = state.clone();
+        tokio::spawn(
+            async move {
+                acquire_redis_lock_and_fetch_data(&state).await.ok();
```

You can propagate the error inside this method and log it here in one place instead of logging it all over